code/modules/mob/living/basic/vermin/butterfly.dm

```diff
@@ -44,3 +44,55 @@
 
 /mob/living/basic/butterfly/lavaland
 	unsuitable_atmos_damage = 0
+
+/mob/living/basic/butterfly/lavaland/temporary
+	name = "strange butterfly"
+	var/obj/item/mod/core/plasma/lavaland/source = NONE
```

Comment:
If you switch to using signals for everything except distance tracking, you can just cast this as an `atom` instead and then have temporary butterflies which live within a range of any atom.